block/balance.go

```diff
@@ -0,0 +1,46 @@
+package block
+
+import (
+	"context"
+	"github.com/dymensionxyz/dymint/types"
+	"strconv"
+	"time"
+)
+
+const CheckBalancesInterval = 3 * time.Minute
+
+// MonitorBalances checks the balances of the node and updates the gauges for prometheus
+func (m *Manager) MonitorBalances(ctx context.Context) error {
+	ticker := time.NewTicker(CheckBalancesInterval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-ticker.C:
+			m.logger.Info("Checking balances.")
+			balances, err := m.checkBalances()
```

Comment:
yea i understand this, my point is different. imagine this:
1. you get the error from SL get balance, but success in DA
2.  `checkBalances` returns `err` and ideally you want to know why SL returned error
3. but a few lines below in `balance.DA != nil` condition you overwrite the `err` variable when you do `ParseFloat`
4. now you've lost the error from `checkBalances`

that's why i suggest renaming it or checking right away after `checkBalances` (eg logging) to avoid this overwrite.